### ISSUE TITLE
WinAppDriver - Unable to get the parent of the element using xpath

### ISSUE THREAD

**Issue by arpitkup** [microsoft/WinAppDriver#1452]
Hi, I am doing a proof of concept to automate WPF application using WinAppDriver. I have situation where a button does not have a unique id, name or classname (or any other properties that can be identified uniquely). However, there is a child element available to this button that has a text that is unique. 

So I am trying to locate the child element (which I can) and then attempting to click on a parent (which is a button I want to click on) using below code but I am getting "An element could not be located on the page using the given search parameters." error. 

 It looks like the xpath does not work the same way as it works with web applications using selenium. Can someone please help with this ? Below is what I tried. 

1) 
var eleRace = ele.FindElementByXPath(".//Text[@Name='LocatorName']/..");

2)
var eleRace = ele.FindElementByXPath(".//Text[@Name='LocatorName']/parent::*");

3)
           var ele = driver.FindElementByAccessibilityId("MainUserControl");
           var eleOne = ele.FindElementByXPath(".//Text[@Name='LocatorName']");
           var eleTwo = eleOne .FindElement(By.XPath("./.."));

![Capture](https://user-images.githubusercontent.com/46307606/108457210-18728880-72ad-11eb-8807-f859bec161f8.PNG)

**Comment by anunay1** [comment]
use the below code: 
driver.findelementsbytagname("Button").findelementsbyxpath("*/*")

findelementsbyxpath this will give all the child elements and check for the child element and using the index click on the button click the button

and the driver.findelementsbytagname("Button")[index].click():

Please note that this will be slow and finding element by xpath is slow and a known issue.

**Comment by arpitkup** [comment]
Thanks @anunay1. Tried the solution you suggested. driver.findelementsbytagname("Button")..findelementsbyxpath("/") is not returning the child elements. I am getting timeout errors.

**Comment by anunay1** [comment]
try this driver.findelementsbytagname("Button").findelementsbyxpath("*/*")

**Comment by arpitkup** [comment]
yeah that's what I tried. Sorry there was an extra . in my previous comment.

**Comment by anunay1** [comment]
What happens if you directly try driver.findelemenbybyname("Races")

**Comment by arpitkup** [comment]
I am getting "element is not a pointer - or keyboard interactable" error

**Comment by arpitkup** [comment]
> What happens if you directly try driver.findelemenbybyname("Races")

**Comment by liljohnak** [comment]
Would something like this work?`.FindElementsByClassName("Button").First(elm elm.FindElementsByName("Races").Any());`

**Comment by Shakevg** [comment]
@Could you try next code:
``` cSharp
var container = driver.FindElementByAccessibilityId("MainUserControl");
var elemToClick = container.FindElementByXPath(".//Text[@Name='Races']/ancestor::Button[1]");
elemToClick .Click();
```

If it does not work maybe Inspector.exe is not showing a real page source. Coul you save Pages source and post screen with particular elements here.
